test/e2e/framework/ginkgowrapper.go

```diff
@@ -31,3 +37,114 @@ func AnnotatedLocation(annotation string) types.CodeLocation {
 	codeLocation = types.NewCustomCodeLocation(annotation + " | " + codeLocation.String())
 	return codeLocation
 }
+
+// It is a wrapper around ginkgo.It which supports [WithFeatureGate],
+// [WithFeature], [WithEnvironment], [WithConformance], [WithNodeConformance]
+// as optional arguments in addition to those already supported by ginkgo
+// itself, like [ginkgo.Label].
+func It(text string, args ...interface{}) {
+	var ginkgoArgs []interface{}
+	var offset ginkgo.Offset
+
+	addLabel := func(label string) {
+		text += fmt.Sprintf(" [%s]", label)
+		ginkgoArgs = append(ginkgoArgs, ginkgo.Label(label))
+	}
+
+	for _, arg := range args {
+		switch arg := arg.(type) {
+		case withFeature:
+			addLabel("Feature:" + string(arg))
+		case withFeatureGate:
+			addLabel("FeatureGate:" + string(arg.featureGate))
+			addLabel(arg.level)
+		case withEnvironment:
+			addLabel("Environment:" + string(arg))
+		case withConformance:
+			addLabel("Conformance")
+		case withNodeConformance:
+			addLabel("NodeConformance")
+		case withDisruptive:
+			addLabel("Disruptive")
+		case withSerial:
+			addLabel("Serial")
+			ginkgoArgs = append(ginkgoArgs, ginkgo.Serial)
+		case ginkgo.Offset:
+			offset = arg
+		default:
+			ginkgoArgs = append(ginkgoArgs, arg)
+		}
+	}
+
+	offset++
+	ginkgoArgs = append(ginkgoArgs, offset)
+	ginkgo.It(text, ginkgoArgs...)
+}
+
+// TODO: Describe and Context wrappers.
+
+func WithFeature(name string) withFeature {
+	// TODO: verify that the feature is known.
+	return withFeature(name)
+}
+
+// WithFeatureGate adds a certain feature gate as label to a [framework.It].
+// The feature gate must be listed in
+// [k8s.io/apiserver/pkg/util/feature.DefaultMutableFeatureGate].
```

Comment:
this will solve the problem of feature promotion, that is common to left e2e tests behind and not update them ... I think is something we should do